mantidproject/mantid
# Enforce synchronicity in Muon Analysis

**Description of work.**

The Observerable class in the Muon interface was set up to always invoke calls on the GUI thread so that paint events could occur. It was not enforcing synchronicity in this cal however which led to problems when multiple workspaces were deleted from the ADS at once. I have updated the Observer to use the QAppThreadCall class which I have moved out of the workbench into the generic qt utils folder in order to ensure that calls are always made synchronously on the correct thread.

**To test:**
  * Open the muon analysis 2 GUI, and load a large number of runs, delete them all at once and check the GUI handles this gracefully. 
  * Check that all the matplotlb functionality still works. i.e. that I havn't broken any import by moving the QAppThreadCall class.



*There is no associated issue.*

*This does not require release notes* because this corrects a bug introduced this release


---

#### Reviewer ####

Please comment on the following ([full description](http://developer.mantidproject.org/IndividualTicketTesting/)):

##### Code Review #####

- Is the code of an acceptable quality?
- Does the code conform to the [coding standards](http://developer.mantidproject.org/Standards/)?
- Are the unit tests small and test the class in isolation?
- If there are changes in the release notes then do they describe the changes appropriately?

##### Functional Tests #####

- Do changes function as described? Add comments below that describe the tests performed?
- Do the changes handle unexpected situations, e.g. bad input?
- Has the relevant (user and developer) documentation been added/updated?

Does everything look good? Mark the review as **Approve**. A member of `@mantidproject/gatekeepers` will take care of it.
